sqlmesh/core/plan/builder.py

```diff
@@ -524,6 +537,48 @@ def _adjust_new_snapshot_intervals(self) -> None:
             if new.is_forward_only:
                 new.dev_intervals = new.intervals.copy()
 
+    def _get_schema_changes_for_snapshot(self, snapshot: Snapshot) -> t.Optional[t.List[exp.Alter]]:
+        """Get schema changes for a snapshot if available, or None if schema analysis can't be performed.
+
+        IMPORTANT: This method applies IGNORE filtering, so returned expressions will have
+        ignored change types already filtered out. This ensures that when IGNORE is set,
+        no alter expressions are generated for the ignored change types.
+        """
+        if snapshot.name not in self._context_diff.modified_snapshots:
+            return None
+
+        new, old = self._context_diff.modified_snapshots[snapshot.name]
+
+        # we must know all columns_to_types to determine schema changes
+        old_columns_to_types = old.model.columns_to_types or {}
+        new_columns_to_types = new.model.columns_to_types or {}
+
+        if not (
+            columns_to_types_all_known(old_columns_to_types)
+            and columns_to_types_all_known(new_columns_to_types)
+        ):
+            return None
+
+        # Generate all schema changes first
+        alter_expressions = get_schema_differ(snapshot.model.dialect).compare_columns(
+            new.name,
+            old_columns_to_types,
+            new_columns_to_types,
+        )
+
+        # Apply IGNORE filtering BEFORE returning expressions
```

Comment:
This feels redundant. I think if ignore is set we shouldn't even iterate over the list. I don't think this method should be concerned with this.